examples/spark/lgbm-parallel.py

```diff
@@ -0,0 +1,241 @@
+import functools
+import logging
+import os
+import pickle
+import shutil
+from multiprocessing.pool import ThreadPool
+from typing import Tuple, List, Dict, Any
+
+# noinspection PyUnresolvedReferences
+from pyspark import inheritable_thread_target
+from pyspark.ml.feature import VectorAssembler
+from pyspark.sql import DataFrame, SparkSession
+from pyspark.sql import functions as sf
+from synapse.ml.lightgbm import LightGBMClassifier, LightGBMRegressor
+
+from examples.spark.examples_utils import get_spark_session, get_dataset_attrs, prepare_test_and_train
+from sparklightautoml.dataset.base import SparkDataset
+from sparklightautoml.pipelines.features.lgb_pipeline import SparkLGBSimpleFeatures
+from sparklightautoml.reader.base import SparkToSparkReader
+from sparklightautoml.tasks.base import SparkTask
+from sparklightautoml.transformers.scala_wrappers.balanced_union_partitions_coalescer import \
+    BalancedUnionPartitionsCoalescerTransformer
+from sparklightautoml.utils import log_exec_timer
+
+logger = logging.getLogger(__name__)
+
+
+params = {
+    'learningRate': 0.01,
+    'numLeaves': 32,
+    'featureFraction': 0.7,
+    'baggingFraction': 0.7,
+    'baggingFreq': 1,
+    'maxDepth': -1,
+    'minGainToSplit': 0.0,
+    'maxBin': 255,
+    'minDataInLeaf': 5,
+    'numIterations': 3000,
+    'earlyStoppingRound': 200,
+    'objective': 'binary',
+    'metric': 'auc'
+}
+
+
+class ParallelExperiment:
+    def __init__(self, spark: SparkSession, dataset_name: str):
+        self.spark = spark
+        self.dataset_name = dataset_name
+        self.partitions_num = 4
+        self.base_dataset_path = f"/opt/spark_data/parallel_slama_{dataset_name}"
+        self.train_path = os.path.join(self.base_dataset_path, "train.parquet")
+        self.test_path = os.path.join(self.base_dataset_path, "test.parquet")
+        self.metadata_path = os.path.join(self.base_dataset_path, "metadata.pickle")
+
+    def prepare_dataset(self, force=True):
+        logger.info(f"Preparing dataset {self.dataset_name}. "
+                    f"Writing train, test and metadata to {self.base_dataset_path}")
+
+        if os.path.exists(self.base_dataset_path) and not force:
+            logger.info(f"Found existing {self.base_dataset_path}. Skipping writing dataset files")
+            return
+        elif os.path.exists(self.base_dataset_path):
+            logger.info(f"Found existing {self.base_dataset_path}. "
+                        f"Removing existing files because force is set to True")
+            shutil.rmtree(self.base_dataset_path)
+
+        seed = 42
+        cv = 5
+        path, task_type, roles, dtype = get_dataset_attrs(self.dataset_name)
+
+        train_df, test_df = prepare_test_and_train(self.spark, path, seed)
```

Comment:
couldn't find usual params like test_size, stratify, etc. Was it intended

Reply:
added test_size, though it is not for the examples.
stratify is not applicable here.